### ISSUE TITLE
OpenSSL used in Docker base image is affected by CVE-2023-0286, though it's not used by Typesense server

### ISSUE THREAD

**Issue by wegerts** [typesense/typesense#895]
## Description

Just scanned the image on Docker Hub and found a high severity caused by an outdated base image:

```bash
✗ High severity vulnerability found in openssl/libssl1.1
  Description: CVE-2023-0286
  Info: https://security.snyk.io/vuln/SNYK-UBUNTU2004-OPENSSL-3314788
  Introduced through: ca-certificates@20211016~20.04.1
  From: ca-certificates@20211016~20.04.1 > openssl@1.1.1f-1ubuntu2.16 > openssl/libssl1.1@1.1.1f-1ubuntu2.16
  From: ca-certificates@20211016~20.04.1 > openssl@1.1.1f-1ubuntu2.16
  Fixed in: 1.1.1f-1ubuntu2.17



Package manager:   deb
Project name:      docker-image|typesense/typesense
Docker image:      typesense/typesense:0.24.1.rc3
Platform:          linux/amd64
Base image:        ubuntu:20.04

Tested 96 dependencies for known vulnerabilities, found 25 vulnerabilities.

Your base image is out of date
1) Pull the latest version of your base image by running 'docker pull ubuntu:20.04'
2) Rebuild your local image
```

## Steps to reproduce

```bash
docker scan typesense/typesense:0.24.0
docker scan typesense/typesense:0.24.1.rc3
```

## Expected Behavior

No high severity findings.

## Actual Behavior

One high severity finding, solvable by base image update.

## Metadata

**Typesense Version**: 0.24.0 and 0.24.1.rc3

**OS**: Ubuntu 20.04 LTS

**Comment by kishorenc** [comment]
@wegerts 

Thank you for reporting this. 

To clarify: Typesense server binary is linked statically to a newer version of OpenSSL that's not affected by this CVE. So, Typesense does **not** use the other openssl version present on the image. Nevertheless, we will still update the base image to take care of this, going forward.

**Comment by wegerts** [comment]
@kishorenc Thanks for the fast response and the explanation!
Wanna use the image in a new project but need to pass the security checks to get approval...

**Comment by kishorenc** [comment]
@wegerts 

Here's a build with a newer base image:  `typesense/typesense:0.24.1.rc4`

**Comment by kishorenc** [comment]
Here's a build with a newer base image: `typesense/typesense:0.24.1.rc4`

**Comment by kishorenc** [comment]
@wegerts 

Let me know if this image passes your checks.

**Comment by wegerts** [comment]
@kishorenc Not yet - current report still shows the CVE
```bash
✗ High severity vulnerability found in openssl/libssl3
  Description: CVE-2023-0286
  Info: https://security.snyk.io/vuln/SNYK-UBUNTU2204-OPENSSL-3314792
  Introduced through: openssl/libssl3@3.0.2-0ubuntu1.7, ca-certificates@20211016ubuntu0.22.04.1, adduser@3.118ubuntu5
  From: openssl/libssl3@3.0.2-0ubuntu1.7
  From: ca-certificates@20211016ubuntu0.22.04.1 > openssl@3.0.2-0ubuntu1.8 > openssl/libssl3@3.0.2-0ubuntu1.7
  From: adduser@3.118ubuntu5 > shadow/passwd@1:4.8.1-2ubuntu2.1 > pam/libpam-modules@1.4.0-11ubuntu2.1 > libnsl/libnsl2@1.3.0-2build2 > libtirpc/libtirpc3@1.3.2-2ubuntu0.1 > krb5/libgssapi-krb5-2@1.19.2-2ubuntu0.1 > krb5/libkrb5-3@1.19.2-2ubuntu0.1 > openssl/libssl3@3.0.2-0ubuntu1.7
  Image layer: 'apt-get -y install ca-certificates'
  Fixed in: 3.0.2-0ubuntu1.8



Package manager:   deb
Project name:      docker-image|typesense/typesense
Docker image:      typesense/typesense:0.24.1.rc4
Platform:          linux/amd64

Tested 104 dependencies for known vulnerabilities, found 18 vulnerabilities.
``` 

Closing for now as it's related to the base image/ca-certs package dependencies but I'd highly recommend to 
- Keep an eye on it
- Update the dev image to avoid vulnerable/outdated libs being used for building the static binary
